src/model/EnergyManagementSystemActuator.cpp

```diff
@@ -108,51 +101,78 @@ namespace model {
+      return getObject<ModelObject>().getModelObjectTarget<ThermalZone>(OS_EnergyManagementSystem_ActuatorFields::ZoneorSpaceName);
+    }
+
+    boost::optional<Space> EnergyManagementSystemActuator_Impl::space() const {
+      return getObject<ModelObject>().getModelObjectTarget<Space>(OS_EnergyManagementSystem_ActuatorFields::ZoneorSpaceName);
+    }
+
+    bool EnergyManagementSystemActuator_Impl::setThermalZone(const ThermalZone& thermalZone) {
+      if (auto comp_ = actuatedComponent()) {
+        if (comp_->optionalCast<SpaceLoadInstance>()) {
+          return setPointer(OS_EnergyManagementSystem_ActuatorFields::ZoneorSpaceName, thermalZone.handle());
+        }
+      }
+      LOG(Warn, "For " << briefDescription() << " cannot setThermalZone has it does not have an attached Component Type that is a SpaceLoadInstance");
+      return false;
+    }
+
+    bool EnergyManagementSystemActuator_Impl::setSpace(const Space& space) {
+      if (auto comp_ = actuatedComponent()) {
+        if (comp_->optionalCast<SpaceLoadInstance>()) {
+          return setPointer(OS_EnergyManagementSystem_ActuatorFields::ZoneorSpaceName, space.handle());
+        }
+      }
+      LOG(Warn, "For " << briefDescription() << " cannot setSpace has it does not have an attached Component Type that is a SpaceLoadInstance");
+      return false;
+    }
+
+    void EnergyManagementSystemActuator_Impl::resetZoneOrSpace() {
+      const bool result = setString(OS_EnergyManagementSystem_ActuatorFields::ZoneorSpaceName, "");
       OS_ASSERT(result);
     }
 
   }  // namespace detail
 
-  EnergyManagementSystemActuator::EnergyManagementSystemActuator(const ModelObject& modelObject, const std::string& actuatedComponentType,
+  EnergyManagementSystemActuator::EnergyManagementSystemActuator(const ModelObject& actuatedComponent, const std::string& actuatedComponentType,
                                                                  const std::string& actuatedComponentControlType)
-    : ModelObject(EnergyManagementSystemActuator::iddObjectType(), modelObject.model()) {
+    : ModelObject(EnergyManagementSystemActuator::iddObjectType(), actuatedComponent.model()) {
     OS_ASSERT(getImpl<detail::EnergyManagementSystemActuator_Impl>());
-    /** Do not use this constructor for spaceloads that are defined in SpaceType objects
-  **/
-    bool ok = setActuatedComponent(modelObject);
+    /** Do not use this constructor for spaceloads that are defined in SpaceType objects **/
+
+    if (actuatedComponent.optionalCast<SpaceLoadInstance>()) {
+      LOG(Warn, "When instantiating an EnergyManagementSystemActuator with an actuated component that is a SpaceLoadInstance, you should instead use "
+                "the overloaded constructor that takes a Space (or a ThermalZone if you plan on disabling the Space Feature during "
+                "ForwardTranslation) as the last parameter");
+    }
```

Comment:
When calling the ctor that doesn't set Space / ThermalZone with a SpaceLoadInstance, you get a warn.